Python basic module execution unit test on delegation of backend_with_compiler_demo

Amyh11325 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#60468 Python basic module execution unit test on delegation of backend_with_compiler_demo**

Summary:
Added a unit test for the execution of a basic module with a compiler

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v returns a successful test

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

Differential Revision: [D29306225](https://our.internmc.facebook.com/intern/diff/D29306225)

Commit:
Python basic module execution unit test on delegation of backend_with_compiler_demo

Summary:
Added a unit test for the execution of a basic module with a compiler

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v returns a successful test

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29306225).

iseeyuan commented on test/jit/test_backends.py:
```diff
@@ -465,3 +465,126 @@ def test_save_load(self):
+        torch_root = Path(__file__).resolve().parent.parent.parent
+        p = torch_root / 'build' / 'lib' / 'libbackend_with_compiler.so'
+        torch.ops.load_library(str(p))
+        # torch.ops.load_library("//caffe2/test/cpp/jit:test_backend_compiler")
+        # Subclasses are expected to set up three variables in their setUp methods:
+        # module - a regular, Python version of the module being tested
+        # scripted_module - a scripted version of module
+        # lowered_modle - a version of module lowered to a backend
```
Remove the commented lines?

Amyh11325 replied:
Including the bottom 4? I just copied those from the existing unit test.

iseeyuan replied:
Good point. The bottom 4 should be kept.

iseeyuan commented on test/jit/test_backends.py:
```diff
@@ -465,3 +465,126 @@ def test_save_load(self):
         self.selective_lowering_test.test_errors()
+
+
+# Unit Tests for backend with compiler
```
Add more comments here: 
"The reason to add it as a separate test case, than reusing the existing TestBackends, is these two cases cover different aspects. The actually backend implementation in this test is different. It has a simple demo compiler to test the end-to-end flow. However, this test cannot cover the selective_lowering for now, which is covered in TestBackends."

iseeyuan commented on test/jit/test_backends.py:
```diff
@@ -465,3 +465,126 @@ def test_save_load(self):
+        """
+        Save and load the lowered module.
+        """
+        self.lowered_module = self.getExportImportCopy(self.lowered_module)
```
Is this test actually called? Could we add it to TestBackendsWithCompiler()?

iseeyuan reviewed (changes requested):
It's a great PR in general! Added some comments inline. Please add save_load tests and update comments. Thanks!

Commit:
Update on "Python basic module execution unit test on delegation of backend_with_compiler_demo"


Summary:
Added a unit test for the execution of a basic module with a compiler

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v returns a successful test

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

Differential Revision: [D29306225](https://our.internmc.facebook.com/intern/diff/D29306225)

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29306225).

Commit:
Update on "Python basic module execution unit test on delegation of backend_with_compiler_demo"


Summary:
Added a unit test for the execution of a basic module with a compiler

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v returns a successful test

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

Differential Revision: [D29306225](https://our.internmc.facebook.com/intern/diff/D29306225)

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29306225).

iseeyuan commented on test/jit/test_backends.py:
```diff
@@ -465,3 +466,121 @@ def test_save_load(self):
+Unit Tests for backend with compiler
+This test case and the existing TestBackends are separate because they cover different aspects.
+The actual backend implementation in this test is different.
+It has a simple demo compiler to test the end-to-end flow.
```
```suggestion
It has a simple demo compiler to test the end-to-end flow in mobile.
```

iseeyuan replied:
Could you also apply clang-format on this PR? You could use `clang-format -i test/jit/test_backends.py` from devserver.

iseeyuan reviewed (approved):
LGTM! Please update the comment and apply clang-format (suggested inline) before landing this PR. Thanks!

Commit:
Update on "Python basic module execution unit test on delegation of backend_with_compiler_demo"


Summary:
Added a unit test for the execution of a basic module with a compiler

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v returns a successful test

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

Differential Revision: [D29306225](https://our.internmc.facebook.com/intern/diff/D29306225)

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29306225).

Commit:
Update on "Python basic module execution unit test on delegation of backend_with_compiler_demo"


Summary:
Added a unit test for the execution of a basic module with a compiler

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v returns a successful test

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

Differential Revision: [D29306225](https://our.internmc.facebook.com/intern/diff/D29306225)

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29306225).

Commit:
Update on "Python basic module execution unit test on delegation of backend_with_compiler_demo"


Summary:
Added a unit test for the execution of a basic module with a compiler

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v returns a successful test

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

Differential Revision: [D29306225](https://our.internmc.facebook.com/intern/diff/D29306225)

[ghstack-poisoned]
